src/test/HostActivationTests/GivenThatICareAboutStartupHooks.cs

```diff
@@ -0,0 +1,622 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.StartupHooks
+{
+    public class GivenThatICareAboutStartupHooks : IClassFixture<GivenThatICareAboutStartupHooks.SharedTestState>
+    {
+        private SharedTestState sharedTestState;
+        private string startupHookVarName = "DOTNET_STARTUP_HOOKS";
+
+        public GivenThatICareAboutStartupHooks(GivenThatICareAboutStartupHooks.SharedTestState fixture)
+        {
+            sharedTestState = fixture;
+        }
+
+        // Run the app with a startup hook
+        [Fact]
+        public void Muxer_activation_of_StartupHook_Succeeds()
+        {
+            var fixture = sharedTestState.PreviouslyPublishedAndRestoredPortableAppProjectFixture.Copy();
+            var dotnet = fixture.BuiltDotnet;
+            var appDll = fixture.TestProject.AppDll;
+
+            var startupHookFixture = sharedTestState.PreviouslyPublishedAndRestoredStartupHookProjectFixture.Copy();
+            var startupHookDll = startupHookFixture.TestProject.AppDll;
+
+            // Simple startup hook
+            dotnet.Exec(appDll)
+                .EnvironmentVariable(startupHookVarName, startupHookDll)
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello from startup hook!")
```

Comment:
Spoke offline. I am still not a fan, but since the core-setup infrastructure doesn't seem to test the return values of the exe this approach is the style of the repo - boo.